### PR TITLE
spectest: precise assertion on (canonical,arithmetic) NaNs

### PR DESCRIPTION
**Pull request by mathetake**
This commit fixes interpreter and spectests so that we precisely match the 
semantics around NaN values. Previously, we round all the NaN values into
`math.NaN()` and only assert with `math.IsNaN` in the spectests. But now
we check the underlying bit pattern of NaN values and do assertion while 
differentiating arithmetic NaNs vs the canonical NaN.

Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by mathetake** [comment]
whew glad that we don't need to modify compiler at all!

**Comment by mathetake** [comment]
lol on again Go has different semantics for math.Trunc, Nearest, Ceil and Floor on RISCV vs others

**Comment by mathetake** [comment]
almost all changes are in the re-generation of spectests so nothing special most of lines 😄

File: .github/workflows/spectest.yaml [hunk 1]
```diff
@@ -92,7 +92,7 @@ jobs:
       - name: Build scratch container
         run: |
           echo 'FROM scratch' >> Dockerfile
-          echo 'CMD ["/test", "-test.v"]' >> Dockerfile
```
Comment (mathetake) [inline .github/workflows/spectest.yaml:95]: get rid of unnecessary verbosity :D

**Review by codefromthecrypt — APPROVED** [review]
great! less perl is best :D

**Pull request merged**